src/tests/multihost/ad/test_adparameters_ported.py

```diff
@@ -0,0 +1,1749 @@
+import random
+import re
+import pytest
+
+from sssd.testlib.common.utils import sssdTools
+from sssd.testlib.common.utils import SSSDException
+from sssd.testlib.common.utils import ADOperations
+
+
+@pytest.fixture(scope="function")
+def create_plain_aduser_group(session_multihost, request):
+    """ Create AD user and group without posix attributes"""
+    unique_num = random.randint(9999, 999999)
+    ad_user = 'plainuser%d' % unique_num
+    ad_group = 'plaingroup%d' % unique_num
+
+    password = session_multihost.ad[0].ssh_password
+    domainname = session_multihost.ad[0].domainname.upper()
+    # Create user
+    session_multihost.client[0].run_command(
+        f'adcli create-user {ad_user} -D {domainname} --display-name='
+        f'"Plain {ad_user}"', stdin_text=password, raiseonerr=False
+    )
+
+    # Create group
+    session_multihost.client[0].run_command(
+        f'adcli create-group {ad_group} -D {domainname} -z '
+        f'"Plain {ad_group}"', stdin_text=password, raiseonerr=False
+    )
+
+    # Add member
+    session_multihost.client[0].run_command(
+        f'adcli add-member -D {domainname} {ad_group} {ad_user}',
+        stdin_text=password, raiseonerr=False
+    )
+
+    def remove_plain_ad_user_group():
+        """ Remove windows AD user and group """
+        ad = ADOperations(session_multihost.ad[0])
+        ad.delete_ad_user_group(ad_group)
+        ad.delete_ad_user_group(ad_user)
+
+    request.addfinalizer(remove_plain_ad_user_group)
+    return ad_user, ad_group
+
+
+@pytest.fixture(scope="class")
+def change_client_hostname(session_multihost, request):
+    """ Change client hostname to a truncated version in the AD domain"""
+    cmd = session_multihost.client[0].run_command('hostname', raiseonerr=False)
+    old_hostname = cmd.stdout_text.rstrip()
+    ad_domain = session_multihost.ad[0].domainname
+    try:
+        new_hostname = session_multihost.client[0].external_hostname.\
+            split('.')[0]
+    except (KeyError, AttributeError):
+        new_hostname = old_hostname.split('.')[0]
+    if new_hostname.startswith('ci-'):
+        new_hostname = new_hostname[3:]
+    new_hostname = new_hostname[:15] + "." + ad_domain
+    session_multihost.client[0].run_command(
+        f'hostname {new_hostname}', raiseonerr=False
+    )
+
+    def restore():
+        """ Restore hostname """
+        session_multihost.client[0].run_command(
+            f'hostname {old_hostname}',
+            raiseonerr=False
+        )
+    request.addfinalizer(restore)
+
+
+@pytest.mark.adparameters
+@pytest.mark.usefixtures("change_client_hostname")
+class TestADParamsPorted():
+    """ BZ Automated Test Cases for AD Parameters ported from bash"""
+
+    @pytest.fixture(autouse=True, scope="class")
+    def _setup(self, session_multihost):
+        self.ad_realm = session_multihost.ad[0].domainname.upper()
+        self.ad_domain_short = self.ad_realm.rsplit('.', 1)[0]
+
+    @staticmethod
+    @pytest.mark.tier1
```

Comment:
add another marker to these tests, ad_parameters, please.

Reply:
Marker @pytest.mark.adparameters is on line 82 for the whole class. It is the same one as in the other ad parameters test in pytest already use. Do You wan to introduce a new one ad_parameters?

Reply:
Actually, let's discuss this in the next team meeting, just leave it for now.